lib/rouge/lexers/solidity.rb

```diff
@@ -0,0 +1,186 @@
+
+        rule(//) { pop! }
+      end
+
+      # :expr_bol is the same as :bol but without labels, since
+      # labels can only appear at the beginning of a statement.
+      state :bol do
+        mixin :expr_bol
+      end
+
+      # TODO: natspec in comments
+      state :inline_whitespace do
+        rule %r/[ \t\r]+/, Text
+        rule %r/\\\n/, Text # line continuation
+        rule %r(/\*), Comment::Multiline, :comment_multi
+      end
+
+      state :whitespace do
+        rule %r/\n+/m, Text, :bol
+        rule %r(//(\\.|.)*?\n), Comment::Single, :bol
+        mixin :inline_whitespace
+      end
+
+      state :expr_whitespace do
+        rule %r/\n+/m, Text, :expr_bol
+        mixin :whitespace
+      end
+
+      state :statements do
+        mixin :whitespace
+        rule %r/(hex)?\"/, Str, :string_double
+        rule %r/(hex)?\'/, Str, :string_single
+        rule %r('(\\.|\\[0-7]{1,3}|\\x[a-f0-9]{1,2}|[^\\'\n])')i, Str::Char
+        rule %r/\d\d*\.\d+([eE]\d+)?/i, Num::Float
+        rule %r/0x[0-9a-f]+/i, Num::Hex
+        rule %r/\d+([eE]\d+)?/i, Num::Integer
+        rule %r(\*/), Error
+        rule %r([~!%^&*+=\|?:<>/-]), Operator
+        rule %r/[()\[\],.]/, Punctuation
+        rule %r/u?fixed\d+x\d+/, Keyword::Reserved
+        rule %r/bytes\d+/, Keyword::Type
+        rule %r/u?int\d+/, Keyword::Type
+        rule id do |m|
+          name = m[0]
+
+          if self.class.keywords.include? name
+            token Keyword
+          elsif self.class.builtins.include? name
+            token Name::Builtin
+          elsif self.class.constants.include? name
+            token Keyword::Constant
+          elsif self.class.keywords_type.include? name
+            token Keyword::Type
+          elsif self.class.reserved.include? name
+            token Keyword::Reserved
+          else
+            token Name
+          end
+        end
+      end
+
+      state :root do
+        mixin :expr_whitespace
+        rule(//) { push :statement }
+        # TODO: function declarations
+      end
+
+      state :statement do
+        rule %r/;/, Punctuation, :pop!
+        mixin :expr_whitespace
+        mixin :statements
+        rule %r/[{}]/, Punctuation
+      end
+
+      state :string_common do
+        rule %r/\\(u[a-fA-F0-9]{4}|x..|[^x])/, Str::Escape
+        rule %r/[^\\\"\'\n]+/, Str
+        rule %r/\\\n/, Str # line continuation
+        rule %r/\\/, Str # stray backslash
+      end
+
+      state :string_double do
+        mixin :string_common
+        rule %r/\"/, Str, :pop!
+        rule %r/\'/, Str
+      end
+
+      state :string_single do
+        mixin :string_common
+        rule %r/\'/, Str, :pop!
+        rule %r/\"/, Str
+      end
+
+       state :comment_multi do
+         rule %r(\*/), Comment::Multiline, :pop!
+         rule %r(/\*), Comment::Multiline, :comment_multi
+         rule %r([^*/]+), Comment::Multiline
+         rule %r([*/]), Comment::Multiline
+       end
```

Comment:
@veox How does this look? This provide a more robust solution, I think (e.g. allows for nested multiline comments). Let me know what you think.

Reply:
It looked simple at first, but then I remembered we shouldn't want to allow nested multi-line comments. :D 

A `/*` within a multi-line is just more characters, and the first encounter of `*/` should close the only level of commenting allowed. That's at least how the Solidity compiler understands it.

Reply:
No problem. Nested comments removed in the latest commit.

Reply:
Cool! This seems to work, too, then. I'll update the sample file anyway with the notes above.